ui/src/Channel.elm

```diff
@@ -298,3 +305,15 @@ updateChannelList model token =
         )
         token
     )
+
+
+updateChannelListForThing : Model -> String -> String -> ( Model, Cmd Msg )
```

Comment:
Wondering whether we can merge `updateChannelList` and `updateChannelListForThing` into one func that receives something like `urlPath` arg instead of `(url.thingsPath ++ [ thingid ] ++ url.channelsPath)` or of `url.channelsPath` and, consequently, remove unnecessary code duplication here

```
    | SubmitOffset String
    | SubmitLimit String
    | SubmitOffsetForThing String String
    | SubmitLimitForThing String String
```

It's just a suggestion and probably will do a PR with this change, but let's keep it like you did it for now. So, for now, this solution LGTM.